unittests/src/main/java/uk/ac/manchester/tornado/unittests/loops/TestLoops.java

```diff
@@ -26,10 +26,117 @@
 
 import uk.ac.manchester.tornado.api.TaskSchedule;
 import uk.ac.manchester.tornado.api.annotations.Parallel;
+import uk.ac.manchester.tornado.api.collections.types.Matrix2DFloat;
 import uk.ac.manchester.tornado.unittests.common.TornadoTestBase;
 
 public class TestLoops extends TornadoTestBase {
 
+    public static void forConstant01(int[] a, final int n) {
+        for (@Parallel int i = 0; i < n; i++) {
+            a[i] = 10;
+        }
+    }
+
+    @Test
+    public void testForConstant01() {
+        final int size = 256;
+        int[] a = new int[size];
+        Arrays.fill(a, 1);
+        new TaskSchedule("s0") //
+                .task("t0", TestLoops::forConstant01, a, size) //
+                .streamOut(a) //
+                .execute();
+        for (int value : a) {
+            assertEquals(10, value);
+        }
+    }
+
+    public static void forConstant02(int[] a, final int n) {
+        for (@Parallel int i = 0; i <= n; i++) {
+            a[i] = 10;
+        }
+    }
+
+    @Test
+    public void testForConstant02() {
+        final int size = 256;
+        int[] a = new int[size];
+        Arrays.fill(a, 1);
+        new TaskSchedule("s0") //
+                .task("t0", TestLoops::forConstant02, a, size) //
+                .streamOut(a) //
+                .execute();
+        for (int value : a) {
+            assertEquals(10, value);
+        }
+    }
+
+    public static void forConstant03(int[] a, int n) {
+        for (@Parallel int i = 0; i < n; i++) {
+            a[i] = 10;
+        }
+    }
+
+    @Test
+    public void testForConstant03() {
+        int size = 256;
+        int[] a = new int[size];
+        Arrays.fill(a, 1);
+        new TaskSchedule("s0") //
+                .task("t0", TestLoops::forConstant03, a, size) //
+                .streamOut(a) //
+                .execute();
+        for (int value : a) {
+            assertEquals(10, value);
+        }
+    }
+
+    public static void forConstant04(Matrix2DFloat m, int n) {
+        for (@Parallel int i = 0; i <= n; i++) {
+            for (@Parallel int j = 0; j <= n; j++) {
+                m.set(i, j, 10);
+            }
+        }
+    }
+
+    @Test
+    public void testForConstant04() {
+        int size = 255;
```

Comment:
We can extend the unit-tests with one test for 2 dimensional array in which each dimension has different size, I tested it and it seems to be working.
e.g. `Matrix2DFloat m = new Matrix2DFloat(256, 64);`

```
task info: s0.t0
	platform          : Intel(R) OpenCL HD Graphics
	device            : Intel(R) Gen9 HD Graphics NEO CL_DEVICE_TYPE_GPU (available)
	dims              : 2
	global work offset: [0, 0]
	global work size  : [64, 256]
	local  work size  : [16, 16, 1]
```